app/main/browser-window.js

```diff
@@ -71,15 +85,39 @@ export async function createBrowserWindow(userAgentClient, onload) {
     browser.webContents.send('user-agent-service-info', { port, host, version });
   });
 
+  // Only check and send the set default browser message if this is the first
+  // window opened, we're in a packaged build, on OSX (for now), and this isn't
+  // already the default browser.
+  const shouldCheckDefaultBrowser = browserWindows.length === 0 &&
+                                    process.env.NODE_ENV === 'production' &&
+                                    process.platform === 'darwin' &&
+                                    !protocols.isDefaultBrowser();
+
   // 'window-ready' is called if an error occurred loading the client, or once
   // the client has connected to the User Agent Service correctly.
   browser.once('window-ready', error => {
+    browser.IS_READY = true;
+
+    if (browser.isDestroyed()) {
+      return;
+    }
+
     // Show this BW (and a devtools window on error).
-    if (!browser.isDestroyed()) {
-      browser.show();
-      if (BUILD_CONFIG.development && error) {
-        browser.openDevTools({ detach: true });
+    browser.show();
+    if (BUILD_CONFIG.development && error) {
+      browser.openDevTools({ detach: true });
+    }
+    // If Tofino is not the default browser
+    if (shouldCheckDefaultBrowser) {
```

Comment:
This isn't setting the default browser, just if it should ask the user. shouldAsk is probably better